### PR TITLE
chore : Stack Analysis E2E tests for Golang Ecosystem

### PR DESCRIPTION
**Pull request by sawood14012**
# Description
Added Golang Stack Analysis E2E tests 

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

 Implemented Scenarios

- [X] Check the golang API without user key
- [X] Check the golang API with invalid user key
- [X] Check the golang API with valid user key
- [X] Check the golang stack analysis v2 for invalid manifest data
- [X] Check the golang stack analysis v2 request and response for proper data
- [X] Data validation checks for golang ecosystem - public vulnerabilities
- [X] Data validation checks for golang ecosystem - private vulnerabilities
- [X] Data validation checks for golang ecosystem - pesudo version

### CODE REVIEW

**Comment by sawood14012** [comment]
[test]

**Comment by sawood14012** [comment]
[test]

**Comment by sawood14012** [comment]
[test]

**Comment by sawood14012** [comment]
[test]

**Comment by sawood14012** [comment]
[test]

**Comment by sawood14012** [comment]
[test]

File: integration-tests/data/golist.json [hunk 1]
```diff
@@ -0,0 +1,1115 @@
+{
+  "version": "v1",
+  "main": "github.com/arajkumar/hello-vuln",
+  "packages": [
```
Comment (prashbnair) [inline integration-tests/data/golist.json:4]: Do we need such a large file for testing. This seems more like a perf test.

File: integration-tests/features/stack_analyses_v2_golang.feature [hunk 2]
```diff
@@ -0,0 +1,128 @@
+Feature: Stack analysis v2 API Golang Ecosystem Tests
+
+
+    @sav2
+    Scenario: Check the golang API without user key
```
Comment (prashbnair) [inline integration-tests/features/stack_analyses_v2_golang.feature:5]: I dont think this is required. The API is is the same. We need not be testing this for all ecosystems.

File: integration-tests/features/stack_analyses_v2_golang.feature [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        Given System is running
+        When I send golang package request with manifest golist.json to stack analysis v2 without valid user key
+        Then I should get 403 status code
+
+
+    @sav2
+    Scenario: Check the golang API with invalid user key
```
Comment (prashbnair) [inline integration-tests/features/stack_analyses_v2_golang.feature:12]: Same as above.

File: integration-tests/features/stack_analyses_v2_golang.feature [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        Given System is running
+        Given Three scale preview service is running
+        When I acquire the use_key for 3scale
+        Then I should get the proper user_key
+        When I send golang package request with manifest valid_golist.json to stack analysis v2 with invalid user key
+        Then I should get 403 status code
+
+    @sav2
+    Scenario: Check the golang API with valid user key
+        Given System is running
```
Comment (prashbnair) [inline integration-tests/features/stack_analyses_v2_golang.feature:22]: same as above

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: integration-tests/data/golist.json (hunk 1 above)
Comment (sawood14012) [reply integration-tests/data/golist.json:4]: will make it short

File: integration-tests/features/stack_analyses_v2_golang.feature (hunk 2 above)
Comment (sawood14012) [reply integration-tests/features/stack_analyses_v2_golang.feature:5]: okay

**Comment by sawood14012** [comment]
[test]

**Review by prashbnair — APPROVED** [review]

**Pull request merged**